File: sert-2014/src/edu/wpi/first/wpilibj/templates/commands/SlowDrive.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package edu.wpi.first.wpilibj.templates.commands;

/**
 *
 * @author SERT
 */
public class SlowDrive extends CommandBase {
    
    public SlowDrive() {
        // Use requires() here to declare subsystem dependencies
        // eg. requires(chassis);
        requires(driveSub);
    }

    // Called just before this Command runs the first time
    protected void initialize() {
    }

    // Called repeatedly when this Command is scheduled to run
    protected void execute() {
    }

    // Make this return true when this Command no longer needs to run execute()
    protected boolean isFinished() {
        return true;
    }

    // Called once after isFinished returns true
    protected void end() {
        driveSub.slowMode();
    }

    // Called when another command which requires one or more of the same
    // subsystems is scheduled to run
    protected void interrupted() {
    }
}

File: sert-2014-C++/Commands/SlowDrive.cpp
#include "SlowDrive.h"

SlowDrive::SlowDrive()
{
	Requires( m_driveSub );
}

// Called just before this Command runs the first time
void SlowDrive::Initialize()
{
}

// Called repeatedly when this Command is scheduled to run
void SlowDrive::Execute() 
{
}

// Make this return true when this Command no longer needs to run execute()
bool SlowDrive::IsFinished() 
{
	return true;
}

// Called once after isFinished returns true
void SlowDrive::End() 
{
	m_driveSub->SlowMode();
}

// Called when another command which requires one or more of the same
// subsystems is scheduled to run
void SlowDrive::Interrupted() 
{
}
